### PR TITLE
Allow negative extents in Yaksa

### PR DESCRIPTION
**Pull request by pavanbalaji**
## Pull Request Description

MPI allows negative extents, but did not allow it.  This PR adds that functionality to Yaksa.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Have read and agree to the Yaksa CLA terms (https://github.com/pmodels/yaksa/wiki/Yaksa-Contributor-License-Agreement)

### CODE REVIEW

**Review by yfguo — APPROVED** [review]
Look good to me!

File: test/dtpools/src/dtpools_attr.c [hunk 1]
```diff
@@ -9,13 +9,13 @@
 #include <inttypes.h>
 
 #define VALUE_FITS_IN_INT(val) ((val) <= INT_MAX)
-#define VALUE_FITS_IN_UINTPTR_T(val) ((val) <= (((uint64_t) 1 << (sizeof(uintptr_t) * 8 - 1)) - 1))
+#define VALUE_FITS_IN_INTPTR_T(val) (((val) * 2) <= (((uint64_t) 1 << (sizeof(uintptr_t) * 8 - 1)) - 1))
```
Comment (hzhou) [inline test/dtpools/src/dtpools_attr.c:12]: I don't this is correct. If `val` is a signed int, `val * 2` may overflow -- in fact, `val` itself may already overflow -- and become negative, which will return `true` here but apparently "does not fit", right?

Comment (pavanbalaji) [reply test/dtpools/src/dtpools_attr.c:12]: `val` is `uint64_t`.  So, the assumption in the code is that `uint64_t` is large enough to accommodate it.  The problematic type would be `uintptr_t` which can be 32 bits.

File: test/dtpools/src/dtpools_attr.c [hunk 2]
```diff
@@ -58,7 +58,7 @@ static int construct_contig(DTP_pool_s dtp, int attr_tree_depth, DTPI_Attr_s * a
 
         count /= attr->u.contig.blklen;
 
-        if (VALUE_FITS_IN_UINTPTR_T(extent * count))
+        if (VALUE_FITS_IN_INTPTR_T(extent * count))
```
Comment (hzhou) [inline test/dtpools/src/dtpools_attr.c:61]: Following up on your reply (https://github.com/pmodels/yaksa/pull/130/commits/5c4c84e0ebdf7636cb26293dbd211307aa87a535#r519047604), both `extent` and `count` is signed integer now, right?

Comment (pavanbalaji) [reply test/dtpools/src/dtpools_attr.c:61]: `extent` is `int64_t`.  This macro is checking if it fits in `intptr_t`.

Comment (hzhou) [reply test/dtpools/src/dtpools_attr.c:61]: Does `INTPTR_T_MAX * 2` fit inside `intptr_t`? If `INTPTR_T_MAX * 2` overflow to negative, the macro will retrun `true`, right?

Comment (pavanbalaji) [reply test/dtpools/src/dtpools_attr.c:61]: Oh, I don't know what that `* 2` is for.  It should simply compare `val`.

Comment (pavanbalaji) [reply test/dtpools/src/dtpools_attr.c:61]: `If INTPTR_T_MAX * count overflow to negative`

Why is this happening?  It's `int64_t`, not `intptr_t`.

Comment (hzhou) [reply test/dtpools/src/dtpools_attr.c:61]: What if `int64_t` and `intptr_t` is both `long int`?

Comment (pavanbalaji) [reply test/dtpools/src/dtpools_attr.c:61]: Then this:

> val is int64_t. So, the assumption in the code is that int64_t is large enough to accommodate it. The problematic type would be intptr_t which can be 32 bits.

Comment (hzhou) [reply test/dtpools/src/dtpools_attr.c:61]: I see.

**Pull request merged**